sdk/iot/hub/tests/cmocka/main.c

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+#include <stdlib.h>
+
+#include <setjmp.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include <cmocka.h>
+
+#include "test_az_iot_hub_client.h"
+
+int main()
+{
+  int result = 0;
+
+  result += test_iot_sas_token();
```

Comment:
Do you guarantee a test never returns a negative value?  -1 += 1 == SUCCESS which isn't what you want.
Another option if all you care about is reporting 1 or more failed
result |= test...()

Reply:
@ewertons @danewalton do you know the answer? I assume the assert statements guarantee this? 

/cc @jspaith @ericwol-msft FYI

Reply:
It returns the number of failed tests as a positive number. I guess if we have`MAXINT + 1` tests that fail we could have an issue but that’s unlikely.

Reply:
I'm not as worried about the MaxInt case.  My concern is a bug in a test that returns an error code rather than the count, thus leading to a sum that masks failures.  
I agree this isn't an issue if each test case properly protects from that problem.

Reply:
It shouldn't. Each of these functions is returned the value from `cmocka_run_group_tests_name()`. So something would have to be wrong with CMocka for this paradigm to be a problem I believe.